### PR TITLE
fix(cli): linter/formater watches current directory without args

### LINKED ISSUE #12445: --watch flag doesn't work without argument
Trying to run `deno fmt --watch` or `deno lint --watch` doesn't work as expected. Files are formatted/linted once and any changes to the files won't retrigger it. 

This problem goes away if there's an argument (`deno fmt --watch dir/` or `deno lint --watch dir/`)

### PR DESCRIPTION
**Pull request by ah-yu**
Close #12445

### CODE REVIEW

File: cli/fs_util.rs [hunk 1]
```diff
@@ -236,9 +236,6 @@ where
     .filter_map(|i| i.canonicalize().ok())
     .collect();
 
-  let cur_dir = [std::env::current_dir()?];
-  let files = if files.is_empty() { &cur_dir } else { files };
```
[the remark this answers is not in the document]
Comment (ah-yu) [reply cli/fs_util.rs:240]: https://github.com/denoland/deno/blob/d44011a69e0674acfa9c59bd7ad7f0523eb61d42/cli/main.rs#L1201-L1203
coverage command does this check before `collect_files`

I think that `collect_files` return empty list if `files` is empty makes sense

[the remark this answers is not in the document]
Comment (bartlomieju) [reply cli/fs_util.rs:240]: That's okay then

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @ah-yu

**Comment by bartlomieju** [comment]
@ah-yu please rebase

**Comment by ah-yu** [comment]
@bartlomieju done

**Pull request merged**